stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
     pass
 
 
-class ActivityLap(LoadableEntity):
+class ActivityLap(Lap, BackwardCompatibilityMixin, DeprecatedSerializableMixin):
+    _field_conversions = {
+        'elapsed_time': time_interval,
+        'moving_time': time_interval
+    }
 
-    name = Attribute(str, (SUMMARY, DETAILED))  #: Name of lap
```

Comment:
what is happening here? were these items removed from the API? but i see there's a backward compat mixin now on ActivityLab.

Reply:
No, nothing was removed. `ActivityLap` inherits from the `Lap` class in the `strava_model` module (which, in turn, is generated by the bot that checks Strava's API). In the `Lap` class, you'll find all the documented attributes such as `name`, `distance`, etc.

The difference is that we don't need to maintain all these attributes anymore; the bot will do that for us. The only thing we need to do in our extended classes is to customize some behavior and maintain the undocumented attributes (adding these still needs to be done).

Reply:
Ahhh ok thank you again. i appreciate the time youre taking here to explain this all.